MerkleMountainRange/src/merklemountainrange.rs

```diff
@@ -0,0 +1,242 @@
+// it is possible to have more than one binary tree. Every time you have to get the merkle root (the single merkle
+// proof of the whole MMR) you have the bag the peaks of the individual trees, or mountain peaks.
+//
+// Lets take an example of how to construct one. Say you have the following MMR already made:
+//
+//       /\
+//      /  \
+//     /\  /\   /\
+//    /\/\/\/\ /\/\ /\
+//
+// From this we can see we have 3 trees or mountains. We have constructed the largest possible tree's we can.
+// If we want to calculate the merkle route we will bag each of the mountains in the following way
+//
+//          /\
+//         /\ \
+//        /  \ \
+//       /\   \ \
+//      /  \   \ \
+//     /\  /\  /\ \
+//    /\/\/\/\/\/\/\
+//
+// Lets continue thw example, by adding a single object. Our MMR now looks as follows
+//
+//       /\
+//      /  \
+//     /\  /\   /\
+//    /\/\/\/\ /\/\ /\ /
+//
+// We now have 4 mountains. Lets bag and calculate the merkle root again
+//
+//           /\
+//          /\ \
+//         /\ \ \
+//        /  \ \ \
+//       /\   \ \ \
+//      /  \   \ \ \
+//     /\  /\  /\ \ \
+//    /\/\/\/\/\/\/\ \
+//
+//  Lets continue thw example, by adding a single object. Our MMR now looks as follows
+//
+//           /\
+//          /  \
+//         /    \
+//        /      \
+//       /\      /\
+//      /  \    /  \
+//     /\  /\  /\  /\
+//    /\/\/\/\/\/\/\/\
+//
+// Now we only have a single binary tree, we dont have to bag the mountains to calculate the merkle root. This
+// process continues as you add more objects to the MMR.
+//
+//                 /\
+//                /  \
+//               /    \
+//              /      \
+//             /        \
+//            /          \
+//           /            \
+//          /\             \
+//         /\ \            /\
+//        /  \ \          /  \
+//       /\   \ \        /\   \
+//      /  \   \ \      /  \   \
+//     /\  /\  /\ \    /\  /\  /\
+//    /\/\/\/\/\/\/\  /\/\/\/\/\/\
+//
+// Due to the unique way the MMR is constructed we can easily represent the MMR as a list of the nodes, as when
+// adding nodes you only append. Lets take the following MMR and number the nodes in the order we create them.
+//
+//        7
+//       /  \
+//      /    \
+//     3      6
+//    / \    / \
+//   1   2  4   5
+//
+// Looking above at the example of when you create the nodes, you will see the nodes will have been created in the
+// order as they are named. This means we can easily represent them as a list:
+// Height:  0 | 0 | 1 | 0 | 0 | 1 | 2
+// Node:    1 | 2 | 3 | 4 | 5 | 6 | 7
+//
+// Because of the list nature of the MMR we can easily navigate around the MMR using the following formulas:
+// Jump to peer : 2^(H+1) -1
+// find peak : 2^(H+1) -2 where < total elements
+// left down : 2^H
+// right down: -1
+// height : log2(I+2) -1
+// Pleas note that the formulas are for direct indexs in array's meaning the nodes count from 0 and not 1 as in
```

Comment:
- Replace "Pleas", start sentence with "Note"
- Replace "indexs in array's meaning" with "indexes in the array, meaning"